haskell-commands.el

```diff
@@ -745,16 +747,22 @@ inferior GHCi process."
         (message "Nothing to unignore!")
       (cl-loop for file in (haskell-session-get session
                                                 'ignored-files)
-               do (cl-case (read-event
-                            (propertize (format "Set permissions? %s (y, n, v: stop and view file)"
-                                                file)
-                                        'face 'minibuffer-prompt))
-                    (?y
-                     (haskell-process-unignore-file session file)
-                     (setq changed t))
-                    (?v
-                     (find-file file)
-                     (cl-return))))
+               do
+               (haskell-mode-set-interactive-prompt-state)
```

Comment:
This has to be wrapped in elisp version of `bracket` that is [`unwind-protect`](http://www.gnu.org/software/emacs/manual/html_node/elisp/Cleanups.html).